wcomponents-core/src/main/java/com/github/bordertech/wcomponents/WBeanComponent.java

```diff
@@ -169,7 +169,9 @@ public Object getBeanId() {
 	 */
 	@Override
 	public void setBeanProperty(final String propertyName) {
-		getOrCreateComponentModel().setBeanProperty(propertyName);
+		if (!propertyName.equals(getBeanProperty())) {
```

Comment:
What if the propertyName arg is null?

Reply:
Was going to change it to 
`if (propertyName != null && (getBeanProperty() == null || !propertyName.equals(getBeanProperty())))`
but that wouldn't do anything if the propertyName is null. Can/should a bean property ever be set null?

Reply:
I think your first snippet is testing the wrong thing. You want something like (as per your other null tests):

``` java
if ( propertyName != null ...
```

The only safe assumption for any public or protected member is that object arguments may be null and the return value of a function which returns anything other than a primitive type may also be null, so yes the return value of `getBeanProperty()` may be null _unless_ `getBeanProperty()` is final **and** the method is guaranteed to not return `null` (for example if it gets a property off a bean and if it is null then returns a new instance of the original return type, but this is **very** unlikely).

Reply:
I'm not actually using either code snippet above anymore (see below for when both strings are null). For primitive types I'm using `if (arg != getArg())` which is easy as it compares whether it is null or not.

As for Strings, I've been using `!arg.equals(...)` as they're objects and using the `==` operator checks for the same memory reference which is never likely to be the case here, we only want the value.